### PR TITLE
Add support for CAS OAuth.

### PR DESCRIPTION
**Pull request by timniederhausen**
This PR adds support for [CAS](https://www.apereo.org/projects/cas), an open-source single sign-on system.

### CODE REVIEW

File: src/main/java/com/googlesource/gerrit/plugins/oauth/CasApi.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2015 The Android Open Source Project
```
Comment (davido) [inline src/main/java/com/googlesource/gerrit/plugins/oauth/CasApi.java]: 2016

File: src/main/java/com/googlesource/gerrit/plugins/oauth/CasOAuthService.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2015 The Android Open Source Project
```
Comment (davido) [inline src/main/java/com/googlesource/gerrit/plugins/oauth/CasOAuthService.java]: 2016

File: src/main/java/com/googlesource/gerrit/plugins/oauth/CasOAuthService.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    }
+    if (log.isDebugEnabled()) {
+      log.debug("User info response: {}", response.getBody());
+    }
+    JsonElement userJson =
+        OutputFormat.JSON.newGson().fromJson(response.getBody(),
+            JsonElement.class);
+    if (!userJson.isJsonObject()) {
+      throw new IOException(String.format(
+              "Invalid JSON '%s': not a JSON Object", userJson));
```
Comment (davido) [inline src/main/java/com/googlesource/gerrit/plugins/oauth/CasOAuthService.java]: Here and below: non consistent indentation: 4 spaces on line continuation.

File: src/main/java/com/googlesource/gerrit/plugins/oauth/InitOAuth.java [hunk 4]
```diff
@@ -45,6 +47,8 @@
         PLUGIN_SECTION, pluginName + GitHubOAuthService.CONFIG_SUFFIX);
     this.bitbucketOAuthProviderSection = sections.get(
         PLUGIN_SECTION, pluginName + BitbucketOAuthService.CONFIG_SUFFIX);
+    this.casOAuthProviderSection = sections.get(
+            PLUGIN_SECTION, pluginName + CasOAuthService.CONFIG_SUFFIX);
```
Comment (davido) [inline src/main/java/com/googlesource/gerrit/plugins/oauth/InitOAuth.java]: Same as above. On line 49 it was done correct.

**Comment by timniederhausen** [comment]
PTAL, fixed formatting and copyright year.

**Pull request merged**